quickwit-serve/src/cluster_api/rest_handler.rs

```diff
@@ -20,51 +20,48 @@
 use std::convert::Infallible;
 use std::sync::Arc;
 
-use quickwit_cluster::{ClusterError, ClusterService};
+use quickwit_cluster::{Cluster, ClusterState};
 use serde::Deserialize;
 use warp::{Filter, Rejection};
 
 use crate::Format;
 
 /// Cluster handler.
 pub fn cluster_handler(
-    cluster_service: Arc<dyn ClusterService>,
+    cluster: Arc<Cluster>,
 ) -> impl Filter<Extract = impl warp::Reply, Error = Rejection> + Clone {
-    list_members_filter()
-        .and(warp::path::end().map(move || cluster_service.clone()))
-        .and_then(list_members)
+    cluster_state_filter()
+        .and(warp::path::end().map(move || cluster.clone()))
+        .and_then(cluster_state)
 }
 
 /// This struct represents the QueryString passed to
 /// the rest API.
 #[derive(Deserialize, Debug, PartialEq, Eq)]
 #[serde(deny_unknown_fields)]
-struct ListMembersRequestQueryString {
+struct ClusterStateQueryString {
     /// The output format requested.
     #[serde(default)]
     pub format: Format,
 }
 
-fn list_members_filter(
-) -> impl Filter<Extract = (ListMembersRequestQueryString,), Error = Rejection> + Clone {
-    warp::path!("cluster" / "members")
+fn cluster_state_filter(
+) -> impl Filter<Extract = (ClusterStateQueryString,), Error = Rejection> + Clone {
+    warp::path!("cluster" / "state")
```

Comment:
so there is only one route? Why not cluster/

Reply:
True. Let's put cluster and see if we need other routes later.